Build a regression model.

In [27]:
import pandas as pd


In [87]:
data = pd.read_csv("c:/users/patel/join_data.csv")

In [88]:
data.head()

bike_station  renting   returning  free_bikes  \
0  Ravenswood Ave & Lawrence Ave         1          1           3   
1        Lake Park Ave & 56th St         1          1          19   
2       Ada St & Washington Blvd         1          1          13   
3         Ashland Ave & Grace St         1          1           6   
4      Clark St & Wrightwood Ave         1          1           7   

   total_bikes  longitude_x  latitude_x    category_x  rating  POI_count  \
0            5   -87.674237   41.969090  bike station   4.325         29   
1           21   -87.587782   41.793242  bike station   3.900         20   
2           15   -87.661206   41.882830  bike station   4.075         23   
3            8   -87.668700   41.950687  bike station   4.175         24   
4            9   -87.643118   41.929546  bike station   4.300         25   

   category_y  longitude_y  latitude_y  
0          29   -87.675768   41.967429  
1          20   -87.589782   41.796849  
2          23   -87.657258   41.884466  
3          24   -87.667896   41.948436  
4          25   -87.643348   41.930288

In [89]:
data = data.dropna()

Build a regression model using Python’s `statsmodels` module that demonstrates a relationship between the number of bikes in a particular location and the characteristics of the POIs in that location.

In [90]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 884 entries, 0 to 971
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   bike_station  884 non-null    object 
 1   renting       884 non-null    int64  
 2   returning     884 non-null    int64  
 3   free_bikes    884 non-null    int64  
 4   total_bikes   884 non-null    int64  
 5   longitude_x   884 non-null    float64
 6   latitude_x    884 non-null    float64
 7   category_x    884 non-null    object 
 8   rating        884 non-null    float64
 9   POI_count     884 non-null    int64  
 10  category_y    884 non-null    int64  
 11  longitude_y   884 non-null    float64
 12  latitude_y    884 non-null    float64
dtypes: float64(5), int64(6), object(2)
memory usage: 96.7+ KB


In [78]:
import statsmodels.api as sm

In [91]:
X = data[['rating','POI_count']]
X.describe()

rating   POI_count
count  884.000000  884.000000
mean     3.233362    6.254525
std      1.391453    5.956594
min      0.000000    1.000000
25%      2.666667    2.000000
50%      3.833333    4.000000
75%      4.125000    9.000000
max      5.000000   29.000000

In [92]:
Y = data['total_bikes']
Y.describe()

count    884.000000
mean       6.404977
std        4.312144
min        0.000000
25%        3.000000
50%        6.000000
75%        9.000000
max       29.000000
Name: total_bikes, dtype: float64

In [93]:
X = sm.add_constant(X) # adding a constant
lin_reg = sm.OLS(Y,X)

In [94]:
model = lin_reg.fit()
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:            total_bikes   R-squared:                       0.106
Model:                            OLS   Adj. R-squared:                  0.104
Method:                 Least Squares   F-statistic:                     52.16
Date:                Sun, 23 Apr 2023   Prob (F-statistic):           3.91e-22
Time:                        15:59:58   Log-Likelihood:                -2496.3
No. Observations:                 884   AIC:                             4999.
Df Residuals:                     881   BIC:                             5013.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          3.2112      0.358      8.982      0.0

Provide model output and an interpretation of the results. 

1.R-squared:In this output, we can see that the value is 0.106. This means that the model is capable of explaining 10.6% of the patterns in the data.

2.P>|t| (or the p-value): This regression output shows a p-value of 0, which means that the probability of the relationship between the average number of bikes and ratings and POI counts of POI being solely due to natural variation is 0. In other words, POI counts and rating does impact the number of bikes.
    
3.coef :we can see that the coef of rating has a value of 0.7317 and 0.1324 . This means that increase in POI count and rating has positive impact on number of bikes.    
    


# Stretch

How can you turn the regression model into a classification model?